source/engine/graphics/core/Instance.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "platform/win32/NcWin32.h"
+
+#include <string_view>
+#include <vector>
+
+#define VK_USE_PLATFORM_WIN32_KHR
+#include <vulkan/vulkan.hpp>
+
+namespace nc::graphics
+{
+class Instance
+{
+    public:
+        Instance(std::string_view appName, uint32_t appVersion,
+                 uint32_t apiVersion, bool enableValidationLayers);
+
+        Instance(const Instance&) = delete;
+        Instance(Instance&&) = delete;
+        Instance& operator=(const Instance&) = delete;
+        Instance& operator=(Instance&&) = delete;
+
+        auto VkInstance() const noexcept -> vk::Instance
+        {
+            return m_instance.get();
+        }
+
+        auto CreateSurface(HWND hwnd, HINSTANCE hinstance) const -> vk::UniqueSurfaceKHR;
+
```

Comment:
Is there intentionally an extra newline here?

Reply:
Nope, fixed.